automation-roles/99-generic/infra/ssh-keygen/tasks/ssh-keygen-rsa.yml

```diff
@@ -0,0 +1,51 @@
+---
+- name: Retrieve SSH public key from the vault
+  include_role: 
+    name: vault-get-secret
+  vars:
+    secret_name: "ocp-ssh-pub-key"
+    secret_group: "{{ environment_name }}"
+    _p_secret_variable: _ocp_ssh_pub_key
+
+- block:
+  - name: Create temporary directory to hold SSH keys
+    tempfile:
+      state: directory
+      suffix: _ssh
+    register: ssh_tempdir
+
+  - name: Create SSH key if not already in the vault and managed
+    community.crypto.openssh_keypair:
+      path: "{{ ssh_tempdir.path }}/id_ssh_rsa"
+      type: rsa
+      comment: "provision-key for id_ssh_rsa"
+
+  - name: Show SSH key file name
+    debug:
+      msg: "{{ ssh_tempdir.path }}/id_ssh_rsa"
+
+  - name: Store SSH private key id_ssh_rsa in vault
+    include_role:
```

Comment:
You need to set the secret specifying an input file. The vault-set-secret role correctly handles the input file. In this case, the final new-line character still gets stripped from the input file.